### PR TITLE
feat: support codeLens.testEnvs

### PR DESCRIPTION
**Pull request by ahuigo**
For https://github.com/denoland/vscode_deno/issues/661

This new new option `codeLens.testEnvs` is used to custom environment variables for **deno test**

### CODE REVIEW

**Review by bartlomieju — COMMENTED** [review]
I think it's reasonable solution, @crowlKats thoughts?

**Comment by crowlKats** [comment]
Yea, think so too. Will test it to see if it works properly

**Comment by bartlomieju** [comment]
@crowlKats any update?

File: README.md [hunk 1]
```diff
@@ -149,6 +149,7 @@ extension has the following configuration options:
 - `deno.codeLens.testArgs`: Provides additional arguments that should be set
   when invoking the Deno CLI test from a code lens. _array of strings, default
   `[ "--allow-all" ]`_.
+- `deno.codeLens.testEnvs`: Provides additional process environments
```
Comment (bartlomieju) [inline README.md:152]: These docs should be updated to provide more information about its purpose

File: docs/testing.md [hunk 2]
```diff
@@ -18,8 +18,14 @@ The code lenses are enabled by default. To disable them, set
...
-`deno.codeLens.testArgs`. They default to `[ "--allow-all" ]`. In addition, when
-executing the test, the extension will reflect the `deno.unstable` setting in
+are supplied when executing the Deno CLI. 
+
+Deno plugin supports these configration:
+
+1. `deno.codeLens.testArgs`. They default to `[ "--allow-all" ]`. 
+2. `deno.codeLens.testEnvs`. This envs will be passed to cli as process enviroments.
+
+## deno.unstable
```
Comment (bartlomieju) [inline docs/testing.md:28]:
```suggestion
## `deno.unstable`

```

**Comment by crowlKats** [comment]
@bartlomieju apologies; I tried however no luck getting it to work

**Comment by bartlomieju** [comment]
> @bartlomieju apologies; I tried however no luck getting it to work

Okay, I'll investigate it myself after NYE

File: README.md (hunk 1 above)
Comment (ahuigo) [reply README.md]: I've provided more information about its purpose.

**Review by bartlomieju — COMMENTED** [review]
I can't get it to work either, putting variables in `deno.codeLens.testEnvs` has no effect on me. @ahuigo how were you able to make it work?

**Comment by ahuigo** [comment]
> I can't get it to work either, putting variables in `deno.codeLens.testEnvs` has no effect on me. @ahuigo how were you able to make it work?

@bartlomieju 

# My test code
```
import { assertEquals } from "https://deno.land/std@0.164.0/testing/asserts.ts";

Deno.test("test deno testEnvs", async (t) => {
  await t.step("test 1", async () => {
    console.error("env ULTRA_MODE:",Deno.env.get('ULTRA_MODE'))
    assertEquals(Deno.env.get('ULTRA_MODE'), 'debug')
  });
});

```
# Reproduce my test 
## 1. Set `testEnvs` 
In my test, I set `ULTRA_MODE` to be `development123`:

image
image

## 2. Then click **Debug** or **Run Test**
In the following figure, we can see that the cli gets the correct value `development123`

image
image

# My device
```
macOsx: 12.6
cpu: Darwin 21.6.0 arm64 core(10) Apple M1 Pro
vscode: 1.71.2 74b1f979648cc44d385a2286793c226e611f59e7 arm64
deno 1.29.1 (release, aarch64-apple-darwin)
v8 10.9.194.5
typescript 4.9.4
deno extensions: denoland.vscode-deno@3.15.0
```

**Comment by bartlomieju** [comment]
@ahuigo when I do the same steps I get following output:

Screenshot 2023-01-04 at 13 37 04

One difference is that I don't see anything in "Debug console" or "Terminal" tabs.

To be honest I'm a bit at loss why that happens. Everything looks correct in the PR.

**Comment by bartlomieju** [comment]
I think something is wrong with my setup - I can't hit any debug breakpoints I set in the client code :/ I'm looking into it

**Comment by ahuigo** [comment]
Try turning off the Deno plugin. @bartlomieju

**Comment by bartlomieju** [comment]
> Try turning off the Deno plugin. @bartlomieju

I tried that, including removing Deno extension completely and it doesn't really help. No breakpoints are ever hit for the "test" command.

**Comment by ahuigo** [comment]
I've encountered that if I set **too many** breakpoints, no breakpoints will be hit.

**Comment by nayeemrmn** [comment]
`deno.env` and `deno.envFile` should work for this now.

**Pull request closed without merging**